conf/scalardl-custom-values.yaml

```diff
@@ -38,7 +38,7 @@ envoy:
                 - key: app.kubernetes.io/name
                   operator: In
                   values:
-                    - scalar-ledger
+                    - scalar-envoy
```

Comment:
This PodAntiAffinity setting does not work because it does not match the label selector.

`scalar-envoy` Pod has the following labels:

```
$ git show -q
commit 3eb6e3fcb9428958d3cf93b817e0fdc57282d33a (HEAD -> fix-pod-antiaffinity, origin/fix-pod-antiaffinity)
Author: Hiroyuki Yamada <mogwaing@gmail.com>
Date:   Wed Feb 3 14:11:09 2021 +0900

    fix
$ helm template charts/stable/scalardl --values=conf/scalardl-custom-values.yaml
...
  template:
    metadata:
      labels:
        app.kubernetes.io/name: scalardl
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/app: envoy
    spec:
      securityContext:
        {}
      restartPolicy: Always
      terminationGracePeriodSeconds: 60
      containers:
        - name: scalardl-envoy
```

You may want to set here as follows:

```diff
diff --git a/conf/scalardl-custom-values.yaml b/conf/scalardl-custom-values.yaml
index 5072342..d08521e 100644
--- a/conf/scalardl-custom-values.yaml
+++ b/conf/scalardl-custom-values.yaml
@@ -38,7 +38,11 @@ envoy:
                 - key: app.kubernetes.io/name
                   operator: In
                   values:
-                    - scalar-envoy
+                    - scalardl
+                - key: app.kubernetes.io/app
+                  operator: In
+                  values:
+                    - envoy
             topologyKey: kubernetes.io/hostname
           weight: 50
```

---

I created a sample repository to see how PodAffinity works (although it was due to a misconfiguration😅). You can try it out if you like.

- https://github.com/superbrothers-sandbox/try-pod-anti-affinity

Reply:
@superbrothers Oh, sorry I thought it was just a typo and didn't really verify it. Thank you.
So, the affinity setting for `scalar-ledger` needs to be updated as well.